File: app/src/main/java/com/example/cykashoppinglist/mapper/TodoMapper.java
package com.example.cykashoppinglist.mapper;

import com.example.cykashoppinglist.entity.Item;
import com.example.cykashoppinglist.entity.TodoEntry;
import com.example.cykashoppinglist.util.DateParser;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class TodoMapper {
	public static Item map(JSONObject jsonObject) throws JSONException {
		TodoEntry entry = new TodoEntry();
		entry.setId(jsonObject.getLong("id"));
		entry.setDate(DateParser.parse(jsonObject.getString("date")));
		entry.setContent(jsonObject.getString("content"));
		return entry;
	}

	public static List<Item> map(JSONArray jsonArray) {
		List<Item> entries = new ArrayList<>();
		for (int i = 0; i < jsonArray.length(); i++) {
			try {
				JSONObject jsonObject = jsonArray.getJSONObject(i);
				entries.add(TodoMapper.map(jsonObject));
			} catch (JSONException e) {
				e.printStackTrace();
			}
		}
		return entries;
	}
}

File: app/src/main/java/com/example/cykashoppinglist/mapper/TodoMapper.kt
package com.example.cykashoppinglist.mapper

import com.example.cykashoppinglist.entity.Item
import com.example.cykashoppinglist.entity.TodoEntry
import com.example.cykashoppinglist.util.DateParser
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

object TodoMapper : Mapper {
	@Throws(JSONException::class)
	override fun toEntity(jsonObject: JSONObject): Item {
		val entry = TodoEntry()
		entry.id = jsonObject.getLong("id")
		entry.date = DateParser.parse(jsonObject.getString("date"))
		entry.name = jsonObject.getString("content")
		return entry
	}

	override fun toEntity(jsonArray: JSONArray): List<Item> {
		val entries: MutableList<Item> = ArrayList()
		for (i in 0 until jsonArray.length()) {
			try {
				val jsonObject = jsonArray.getJSONObject(i)
				entries.add(toEntity(jsonObject))
			} catch (e: JSONException) {
				e.printStackTrace()
			}
		}
		return entries
	}
}